tls/s2n_server_extensions.c

```diff
@@ -175,6 +175,7 @@ int s2n_recv_server_alpn(struct s2n_connection *conn, struct s2n_stuffer *extens
 int s2n_recv_server_status_request(struct s2n_connection *conn, struct s2n_stuffer *extension)
 {
     conn->status_type = S2N_STATUS_REQUEST_OCSP;
+    conn->handshake.handshake_type |= OCSP_STATUS;
```

Comment:
I think the best home for this is`s2n_conn_set_handshake_type` https://github.com/awslabs/s2n/blob/7978d5608662860f47cae7176ab45597841b5776/tls/s2n_handshake_io.c#L193 . We've isolated all of the state machine manipulation to that helper.